lib/stdlib/test/proplists_SUITE.erl

```diff
@@ -77,4 +77,83 @@ examples(_Config) ->
     NegListRes = [{foo, false}, {foo, false}, foo, foo, foo],
     NegListRes = proplists:substitute_negations([{no_foo, foo}], NegList),
 
+    true = #{a => true, b => 1, c => 2} =:= proplists:to_map([a, {b, 1}, {c, 2}, {c, 3}]),
+
+    ok.
+
+map_conversion(_Config) ->
+    %% Simple tests.
+    true = #{} =:= proplists:to_map([]),
+    true = #{a => true, b => true} =:= proplists:to_map([a, b]),
+    true = #{a => true, b => true} =:= proplists:to_map([b, a]),
+    true = #{a => 1, b => true} =:= proplists:to_map([{a, 1}, b]),
+    true = #{a => 1, b => true} =:= proplists:to_map([b, {a, 1}]),
+    true = #{a => 1, b => 2} =:= proplists:to_map([{a, 1}, {b, 2}]),
+    true = #{a => 1, b => 2} =:= proplists:to_map([{b, 2}, {a, 1}]),
+    true = #{b => true} =:= proplists:to_map(["a", b]),
+    true = #{b => true} =:= proplists:to_map([b, "a"]),
+    true = #{b => true} =:= proplists:to_map([{a}, b]),
+    true = #{b => true} =:= proplists:to_map([b, {a}]),
+    true = #{b => true} =:= proplists:to_map([{a, 1, 2}, b]),
+    true = #{b => true} =:= proplists:to_map([b, {a, 1, 2}]),
+
+    %% Ensure that maps:get/3 using the created map yields the same
+    %% results as proplists:get_value/3 on the original proplist does,
+    %% and that proplists:get_value/3 on a proplist created from the
+    %% map yields the same results as proplists:get_value/3 on the
+    %% original proplist, ie they either all return the same `Value',
+    %% or they all return the `Default' given as respective third argument.
+    Default1 = make_ref(),
+    Default2 = make_ref(),
+    Default3 = make_ref(),
+    InList=[a, b, {a, 1}, {}, {a}, {a, 1, 2}, "foo"],
+    lists:foreach(
+        fun (L1) ->
+            LKs = proplists:get_keys(L1),
+            M = proplists:to_map(L1),
+            L2 = proplists:from_map(M),
+            [] = maps:keys(M) -- LKs,
+            [] = proplists:get_keys(L2) -- LKs, 
```

Comment:
Here you could instead compare `maps:keys(M)` with `proplists:get_keys(L2)` for equality, after sorting both.

Reply:
Ok

Reply:
Done